WalletWasabi/Services/WasabiSynchronizer.cs

```diff
@@ -44,19 +38,22 @@ public class WasabiSynchronizer : NotifyPropertyChangedBase, IFeeProvider
 
 		public WasabiSynchronizer(Network network, BitcoinStore bitcoinStore, WasabiClient client)
 		{
-			CreateNew(network, bitcoinStore, client);
+			Network = Guard.NotNull(nameof(network), network);
+			WasabiClient = Guard.NotNull(nameof(client), client);
+			LastResponse = null;
+			BitcoinStore = Guard.NotNull(nameof(bitcoinStore), bitcoinStore);
+			_running = 0;
+			Cancel = new CancellationTokenSource();
 		}
 
-		public WasabiSynchronizer(Network network, BitcoinStore bitcoinStore, Func<Uri> baseUriAction, EndPoint torSocks5EndPoint)
+		public WasabiSynchronizer(Network network, BitcoinStore bitcoinStore, Func<Uri> baseUriAction, EndPoint torSocks5EndPoint) 
+			: this(network, bitcoinStore, client: new WasabiClient(baseUriAction, torSocks5EndPoint))
 		{
-			var client = new WasabiClient(baseUriAction, torSocks5EndPoint);
-			CreateNew(network, bitcoinStore, client);
 		}
 
-		public WasabiSynchronizer(Network network, BitcoinStore bitcoinStore, Uri baseUri, EndPoint torSocks5EndPoint)
+		public WasabiSynchronizer(Network network, BitcoinStore bitcoinStore, Uri baseUri, EndPoint torSocks5EndPoint) :
```

Comment:
In this constructor the semicolon should go before the `this` as in the previous one.

Reply:
Ummm it could be no so easy it seems. Okay, try it and if it requires many changes then ignore this point.

Reply:
Not sure, what you mean here. But maybe it has been already addressed.